Ryujinx.HLE/HOS/Horizon.cs

```diff
@@ -283,25 +303,26 @@ public void LoadKip(string kipFile)
 
             XciPartition securePartition = xci.OpenPartition(XciPartitionType.Secure);
 
-            foreach (DirectoryEntry ticketEntry in securePartition.EnumerateEntries("*.tik"))
+            foreach (DirectoryEntryEx ticketEntry in securePartition.EnumerateEntries("/", "*.tik"))
             {
-                Ticket ticket = new Ticket(securePartition.OpenFile(ticketEntry.FullPath, OpenMode.Read).AsStream());
+                Result rc = securePartition.OpenFile(out IFile ticketFile, ticketEntry.FullPath, OpenMode.Read);
+                if (rc.IsFailure()) continue;
 
-                if (!KeySet.TitleKeys.ContainsKey(ticket.RightsId))
-                {
-                    KeySet.TitleKeys.Add(ticket.RightsId, ticket.GetTitleKey(KeySet));
-                }
+                Ticket ticket = new Ticket(ticketFile.AsStream());
+
+                KeySet.ExternalKeySet.Add(new RightsId(ticket.RightsId), new AccessKey(ticket.GetTitleKey(KeySet)));
             }
 
-            foreach (DirectoryEntry fileEntry in securePartition.EnumerateEntries("*.nca"))
+            foreach (DirectoryEntryEx fileEntry in securePartition.EnumerateEntries("/", "*.nca"))
             {
-                IStorage ncaStorage = securePartition.OpenFile(fileEntry.FullPath, OpenMode.Read).AsStorage();
+                Result rc = securePartition.OpenFile(out IFile ncaFile, fileEntry.FullPath, OpenMode.Read);
```

Comment:
```suggestion
                Result resultCode = securePartition.OpenFile(out IFile ncaFile, fileEntry.FullPath, OpenMode.Read);
```

and add an empty line after.